landscape/package/reporter.py

```diff
@@ -207,6 +211,51 @@ def _apt_update_timeout_expired(self, interval):
         last_update = os.stat(stamp).st_mtime
         return (last_update + interval) < time.time()
 
+    def _is_release_upgrader_running(self):
+        """Detect whether ubuntu-release-upgrader is running.
+
+        This is done by iterating the /proc tree (to avoid external
+        dependencies) and checkign the cmdline and the uid of the process.
+        The assumption is that ubuntu-release-upgrader is something that:
+            * is run by a python interpreter
+            * its first argument starts with '/tmp/ubuntu-release-upgrader-'
+            * is executed by root (effective uid == 0)"""
+        logging.debug("Checking if ubuntu-release-upgrader is running.")
+
+        for cmdline in glob.glob("/proc/*/cmdline"):
+            base = os.path.dirname(cmdline)
+            try:
+                with open(cmdline) as fd:
+                    read = fd.read()
+
+                pid = os.path.basename(os.path.dirname(cmdline))
+
+                cmdline = [f for f in filter(lambda x: x != '',
+                                             read.split("\x00"))]
```

Comment:
I'd drop the lambda for list condition
cmdline = [f for f in read.split("\x00") if f]